extensions-core/kafka-indexing-service/src/main/java/io/druid/indexing/kafka/KafkaIndexTask.java

```diff
@@ -95,13 +126,42 @@
   private final InputRowParser<ByteBuffer> parser;
   private final KafkaTuningConfig tuningConfig;
   private final KafkaIOConfig ioConfig;
+  private final Optional<ChatHandlerProvider> chatHandlerProvider;
+
+  private final Map<Integer, Long> endOffsets = new ConcurrentHashMap<>();
+  private final Map<Integer, Long> nextOffsets = new ConcurrentHashMap<>();
+
+  private ObjectMapper mapper;
 
   private volatile Appenderator appenderator = null;
   private volatile FireDepartmentMetrics fireDepartmentMetrics = null;
-  private volatile boolean startedReading = false;
-  private volatile boolean stopping = false;
-  private volatile boolean publishing = false;
+  private volatile DateTime startTime;
+  private volatile Status status;
```

Comment:
Yes, `status` is only modified by the task runner thread which calls run(). I added a comment.